internal/policy/policy_test.go

```diff
@@ -427,3 +427,142 @@ func TestStateFindDelegationEntry(t *testing.T) {
 		assert.Equal(t, &tuf.Delegation{Name: "4", Paths: []string{"file:1/subpath2/*"}, Terminating: false, Role: tuf.Role{KeyIDs: []string{"157507bbe151e378ce8126c1dcfe043cdd2db96e"}, Threshold: 1}}, entry)
 	})
 }
+
+func TestListRules(t *testing.T) {
+	t.Run("t1", func(t *testing.T) {
+		repo, _ := createTestRepository(t, createTestStateWithPolicy)
+
+		rules, err := ListRules(context.Background(), repo)
+		assert.Nil(t, err)
+		expectedRules := []*DelegationWithDepth{
+			{
+				Delegation: tuf.Delegation{
+					Name:        "protect-main",
+					Paths:       []string{"git:refs/heads/main"},
+					Terminating: false,
+					Custom:      nil,
+					Role: tuf.Role{
+						KeyIDs:    []string{"157507bbe151e378ce8126c1dcfe043cdd2db96e"},
+						Threshold: 1,
+					},
+				},
+				Depth: 0,
+			},
+			{
+				Delegation: tuf.Delegation{
+					Name:        "protect-files-1-and-2",
+					Paths:       []string{"file:1", "file:2"},
+					Terminating: false,
+					Custom:      nil,
+					Role: tuf.Role{
+						KeyIDs:    []string{"157507bbe151e378ce8126c1dcfe043cdd2db96e"},
+						Threshold: 1,
+					},
+				},
+				Depth: 0,
+			},
+			{
+				Delegation: tuf.Delegation{
+					Name:        "gittuf-allow-rule",
+					Paths:       []string{"*"},
+					Terminating: true,
+					Custom:      nil,
+					Role: tuf.Role{
+						KeyIDs:    []string{},
+						Threshold: 1,
+					},
+				},
+				Depth: 0,
+			},
+		}
+		assert.Equal(t, expectedRules, rules)
+	})
+	t.Run("t2", func(t *testing.T) {
+		repo, _ := createTestRepository(t, createTestStateWithDelegatedPolicies)
+
+		rules, err := ListRules(context.Background(), repo)
+
+		assert.Nil(t, err)
+		expectedRules := []*DelegationWithDepth{
+			{
+				Delegation: tuf.Delegation{
+					Name:        "1",
+					Paths:       []string{"file:1/*"},
+					Terminating: false,
+					Custom:      nil,
+					Role: tuf.Role{
+						KeyIDs:    []string{"52e3b8e73279d6ebdd62a5016e2725ff284f569665eb92ccb145d83817a02997"},
+						Threshold: 1,
+					},
+				},
+				Depth: 0,
+			},
+			{
+				Delegation: tuf.Delegation{
+					Name:        "4",
```

Comment:
shouldn't 3 be listed before 4? Note that the order of rules is really important, as a terminating rule that appears first may mean the rule after is ignored.

Reply:
The order has been changed so that the original order is preserved.